### ISSUE TITLE
Bug: Rufo breaks HEREDOC syntax

### ISSUE THREAD

**Issue by ivoanjo** [ruby-formatter/rufo#88]
While looking into #83 I tried rufo on a number of repositories for commonly-used gems and came across a few issues.

This one is a testcase extracted from the [pry](https://github.com/pry/pry) gem.

Example code:

```ruby
puts (<<-HELLO
  hello world
HELLO
)
```

which after running rufo turns into

```ruby
puts (<<-HELLO
  hello world
HELLO)
```

which is now invalid:

```ruby
$ ruby heredoc.rb 
heredoc.rb:3: can't find string "HELLO" anywhere before EOF
heredoc.rb:1: syntax error, unexpected end-of-input, expecting tSTRING_CONTENT or tSTRING_DBEG or tSTRING_DVAR or tSTRING_END
puts (<<-HELLO
              ^
```

**Comment by gingermusketeer** [comment]
Thanks for reporting this ❤️. I have a failing test for this issue here https://github.com/ruby-formatter/rufo/compare/chore/fix-heredoc-issue

**Comment by mjago** [comment]
Fixed with #116
